### PR TITLE
Add "3 characters needed" warning to search

### PR DESCRIPTION
**Pull request by haiyingweng**
## Overview

Add "3 characters needed" warning to search

## Changes Made

- added empty state that shows the 3 characters needed if user types search text less than 3

## Test Coverage

simulator

## Screenshots (delete if not applicable)

<details>

  <summary>Screen Shot Name</summary>

  
  
Simulator Screen Shot - iPhone 11 - 2020-11-27 at 23 32 40

</details>

### CODE REVIEW

File: CourseGrab/ViewControllers/Search/SearchTableViewController.swift [hunk 1]
```diff
@@ -122,6 +125,9 @@ extension SearchTableViewController {
     }
 
     override func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        if 1...2 ~= textField.text?.count ?? 0 {
```
Comment (ywsang) [inline CourseGrab/ViewControllers/Search/SearchTableViewController.swift:128]: woah `~=`?? never seen that before

**Review by ywsang — APPROVED** [review]
looks good!

File: CourseGrab/ViewControllers/Search/SearchTableViewController.swift (hunk 1 above)
Comment (mathewscullin) [reply CourseGrab/ViewControllers/Search/SearchTableViewController.swift:128]: it's soo swifty! 💯

**Review by mathewscullin — APPROVED** [review]
awesome stuff!

**Pull request merged**